### PR TITLE
#170766107 users can rate accommodation

### PR DESCRIPTION
**Pull request by Nigorjeanluc**
#### What does this PR do?
This pull request ensure any user can rate accommodation to which he/she had passed a good time recently
### Description of Task to be completed?
 - create `server/middlewares/isUser.js` to verify if the user is signed up.
 - add to `server/middlewares/statusValidation.js` a method to validate if the accommodation rating stars entered by the user.
 - create `server/tests/rating.test.js` file to test this endpoint
#### How should this be manually tested?
You can pull the `develop` or `ft-rate-accommodation-170766107` branch and run these following commands `npm install` and run `npm test` 
#### What are the relevant pivotal tracker stories?
[#170766107](https://www.pivotaltracker.com/n/projects/2429839/stories/170766107)

### Screenshots
screenRate

### CODE REVIEW

**Review by karengir — APPROVED** [review]

**Review by jajajaden01 — APPROVED** [review]

**Review by key-joshua — APPROVED** [review]

**Review by izabayogit — APPROVED** [review]

File: server/middlewares/isUser.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    if (!req.headers.token) {
+      return res.status(401).send({
+        status: 401,
+        error: 'Please provide a token first',
+      });
+    }
+    const { token } = req.headers;
+    const decoded = Auth.decodedToken(token, process.env.SECRET_KEY);
+    req.simpleUser = decoded;
+    if (!await users.userExists('email', req.simpleUser.email)) {
```
Comment (luc-tuyishime) [inline server/middlewares/isUser.js]: Can you refactor this line so that we understand it well?

File: server/tests/rating.test.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    const res = await router().patch('/api/ratings/1')
+      .set('token', invalid).send({ stars: 4 });
+    expect(res.body).to.be.an('object');
+    expect(res.body.status).to.equals(401);
+    expect(res.body.error).to.be.a('string');
+    expect(res.body.error).to.equals('Sign up first please');
+  });
+  it('should rate the acc', async () => {
+    const res = await router().patch('/api/ratings/145447')
+      .set('token', `${otherToken.UnExistUser}gfsdfgdfg`).send({ stars: 4 });
```
Comment (luc-tuyishime) [inline server/tests/rating.test.js]: What does ( gfsdfgdfg ) mean ?

Comment (Nigorjeanluc) [reply server/tests/rating.test.js]: Here I tried to simulate an invalid token. I will store this into a variable

Comment (luc-tuyishime) [reply server/tests/rating.test.js]: Give it a proper name.

Comment (Nigorjeanluc) [reply server/tests/rating.test.js]: Ok It's done now

File: server/controllers/ratingController.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      ratings[2] += 1;
+    } else if (stars === 4) {
+      ratings[3] += 1;
+    } else {
+      ratings[4] += 1;
+    }
+
+    await AccommodationHelper.updateRatings(ratings, parseInt(req.params.id, 10));
+    await RequestHelper.updateRated(parseInt(req.params.id, 10), req.requesterUser.id);
+    const ratedAccommodation = await AccommodationHelper.findAccommodation('id', parseInt(req.params.id, 10));
```
Comment (istafrica-dennw) [inline server/controllers/ratingController.js:42]: Hi @Nigorjeanluc This is too much for a rating. 4 APIs, It is tooooo much. we should be having only one or two maximum. As you could not do it, Let us do it an other way, a table of ratings and do an association of ratings belonging to accommodation.

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]

**Review by istafrica-dennw — APPROVED** [review]

**Pull request merged**